Feature request: limit max octree depth on mesh generation

Odm fails at PoissonRecon step.

my command is
````
docker run -it --rm -v  "/data/my_images/:/code/images" -v "$(pwd)/odm_georeferencing:/code/odm_georeferencing"     -v "$(pwd)/odm_meshing:/code/odm_meshing"     -v "$(pwd)/odm_dem:/code/odm_dem"     -v "$(pwd)/odm_orthophoto:/code/odm_orthophoto"     -v "$(pwd)/odm_texturing:/code/odm_texturing"     -v "$(pwd)/odm_report:/code/odm_report" opendronemap/odm --orthophoto-png --mesh-octree-depth 16 --ignore-gsd --dtm   --smrf-threshold 0.4 --smrf-window 24 --dsm --pc-csv --pc-las --orthophoto-kmz   --ignore-gsd --feature-quality high --max-concurrency 16  --use-hybrid-bundle-adjustment --build-overviews --time --min-num-features 10000
````

Linux  4.19.0-16-amd64 #1 SMP Debian 4.19.181-1 (2021-03-19) x86_64 GNU/Linux

- Docker opendronemap/odm   latest    409551646e8d
  20_04_21_X7_Chambon_All docker image inspect 409551646e8d
        "Id": "sha256:409551646e8d301bcfc1de8bbd0dfba5516ba944366a033dfee18953f337f95c",
        "RepoTags": [
            "opendronemap/odm:latest"
        ],
        "RepoDigests": [
            "opendronemap/odm@sha256:7e54c74fb6b040da60effbdf98980f875337e4b5c8027bc939e3d446703795a7"
        ],
        "Created": "2021-06-22T17:31:39.210863835Z",




````
[INFO]    Merging 52 point cloud chunks to /code/odm_filterpoints/point_cloud.ply
[INFO]    Finished odm_filterpoints stage
[INFO]    Running odm_meshing stage
[INFO]    Writing ODM Mesh file in: /code/odm_meshing/odm_mesh.ply
[INFO]    running "/code/SuperBuild/install/bin/PoissonRecon" --in "/code/odm_filterpoints/point_cloud.ply" --out "/code/odm_meshing/odm_mesh.dirty.ply" --depth 16 --pointWeight 4.0 --samplesPerNode 1.0 --threads 15 -
-maxMemory 94 --bType 2 --linearFit
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
[WARNING] Child returned 134
WARNING] PoissonRecon failed with 7 threads, let's retry with 3...
[INFO]    running "/code/SuperBuild/install/bin/PoissonRecon" --in "/code/odm_filterpoints/point_cloud.ply" --out "/code/odm_meshing/odm_mesh.dirty.ply" --depth 16 --pointWeight 4.0 --samplesPerNode 1.0 --threads 7 --
maxMemory 93 --bType 2 --linearFit
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
----Aborted (core dumped)
[WARNING] Child returned 134
[WARNING] PoissonRecon failed with 3 threads, let's retry with 1...
[INFO]    running "/code/SuperBuild/install/bin/PoissonRecon" --in "/code/odm_filterpoints/point_cloud.ply" --out "/code/odm_meshing/odm_mesh.dirty.ply" --depth 16 --pointWeight 4.0 --samplesPerNode 1.0 --threads 3 --
maxMemory 94 --bType 2 --linearFit
---terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
[WARNING] Child returned 134
[WARNING] PoissonRecon failed with 1 threads, let's retry with 0...
[INFO]    running "/code/SuperBuild/install/bin/PoissonRecon" --in "/code/odm_filterpoints/point_cloud.ply" --out "/code/odm_meshing/odm_mesh.dirty.ply" --depth 16 --pointWeight 4.0 --samplesPerNode 1.0 --threads 1 --
maxMemory 94 --bType 2 --linearFit
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
[WARNING] Child returned 134
[INFO]    running "/code/SuperBuild/install/bin/OpenMVS/ReconstructMesh" -i "/code/odm_meshing/odm_mesh.dirty.ply" -o "/code/odm_meshing/odm_mesh.ply" --remove-spikes 0 --remove-spurious 20 --smooth 0 --target-face-nu
m 400000
10:34:33 [App     ] Build date: Jun 22 2021, 17:22:42
10:34:33 [App     ] CPU: Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz (16 cores)
10:34:33 [App     ] RAM: 125.86GB Physical Memory 975.00MB Virtual Memory
10:34:33 [App     ] OS: Linux 4.19.0-16-amd64 (x86_64)
10:34:33 [App     ] SSE & AVX compatible CPU & OS detected
10:34:33 [App     ] Command line: -i /code/odm_meshing/odm_mesh.dirty.ply -o /code/odm_meshing/odm_mesh.ply --remove-spikes 0 --remove-spurious 20 --smooth 0 --target-face-num 400000

===== Dumping Info for Geeks (developers need this to fix bugs) =====
Child returned 1
Traceback (most recent call last):
  File "/code/stages/odm_app.py", line 89, in execute
    self.first_stage.run()
  File "/code/opendm/types.py", line 340, in run
    self.next_stage.run(outputs)
  File "/code/opendm/types.py", line 340, in run
    self.next_stage.run(outputs)
  File "/code/opendm/types.py", line 340, in run
    self.next_stage.run(outputs)
  [Previous line repeated 3 more times]
  File "/code/opendm/types.py", line 321, in run
    self.process(self.args, outputs)
  File "/code/stages/odm_meshing.py", line 24, in process
    mesh.screened_poisson_reconstruction(tree.filtered_point_cloud,
 File "/code/opendm/mesh.py", line 207, in screened_poisson_reconstruction
    system.run('"{reconstructmesh}" -i "{infile}" '
  File "/code/opendm/system.py", line 106, in run
    raise SubprocessException("Child returned {}".format(retcode), retcode)
opendm.system.SubprocessException: Child returned 1

===== Done, human-readable information to follow... =====

[ERROR]   Uh oh! Processing stopped because of strange values in the reconstruction. This is often a sign that the input data has some issues or the software cannot deal with it. Have you followed best practices for data acquisition? See https://docs.opendronemap.org/flying.html
`````

--- a/opendm/cogeo.py
+++ b/opendm/cogeo.py
@@ -5,7 +5,7 @@ from opendm.concurrency import get_max_memory
 from opendm import io
 from opendm import log
 
-def convert_to_cogeo(src_path, blocksize=256, max_workers=1):
+def convert_to_cogeo(src_path, blocksize=256, max_workers=1, compression="DEFLATE"):
     """
     Guarantee that the .tif passed as an argument is a Cloud Optimized GeoTIFF (cogeo)
     The file is destructively converted into a cogeo.
@@ -30,6 +30,8 @@ def convert_to_cogeo(src_path, blocksize=256, max_workers=1):
         'max_memory': get_max_memory(),
         'src_path': src_path,
         'tmpfile': tmpfile,
+        'compress': compression,
+        'predictor': '2' if compression in ['LZW', 'DEFLATE'] else '1',
     }
 
     try:
@@ -37,7 +39,8 @@ def convert_to_cogeo(src_path, blocksize=256, max_workers=1):
                 "-of COG "
                 "-co NUM_THREADS={threads} "
                 "-co BLOCKSIZE={blocksize} "
-                "-co COMPRESS=deflate "
+                "-co COMPRESS={compress} "
+                "-co PREDICTOR={predictor} "
                 "-co BIGTIFF=IF_SAFER "
                 "-co RESAMPLING=NEAREST "
                 "--config GDAL_CACHEMAX {max_memory}% "

--- a/opendm/config.py
+++ b/opendm/config.py
@@ -277,7 +277,7 @@ def config(argv=None, parser=None):
                               'Default: %(default)s'))
 
     parser.add_argument('--mesh-octree-depth',
-                        metavar='<positive integer>',
+                        metavar='<integer: 1 <= x <= 14>',
                         action=StoreValue,
                         default=11,
                         type=int,

--- a/opendm/orthophoto.py
+++ b/opendm/orthophoto.py
@@ -86,7 +86,7 @@ def post_orthophoto_steps(args, bounds_file_path, orthophoto_file, orthophoto_ti
         generate_orthophoto_tiles(orthophoto_file, orthophoto_tiles_dir, args.max_concurrency)
 
     if args.cog:
-        convert_to_cogeo(orthophoto_file, max_workers=args.max_concurrency)
+        convert_to_cogeo(orthophoto_file, max_workers=args.max_concurrency, compression=args.orthophoto_compression)
 
 def compute_mask_raster(input_raster, vector_mask, output_raster, blend_distance=20, only_max_coords_feature=False):
     if not os.path.exists(input_raster):

--- a/opendm/osfm.py
+++ b/opendm/osfm.py
@@ -193,6 +193,7 @@ class OSFMContext:
                 "optimize_camera_parameters: %s" % ('no' if args.use_fixed_camera_params or args.cameras else 'yes'),
                 "undistorted_image_format: tif",
                 "bundle_outlier_filtering_type: AUTO",
+                "sift_peak_threshold: 0.066",
                 "align_orientation_prior: vertical",
                 "triangulation_type: ROBUST",
                 "retriangulation_ratio: 2",

--- a/stages/odm_app.py
+++ b/stages/odm_app.py
@@ -42,7 +42,7 @@ class ODMApp:
         filterpoints = ODMFilterPoints('odm_filterpoints', args, progress=52.0)
         meshing = ODMeshingStage('odm_meshing', args, progress=60.0,
                                     max_vertex=args.mesh_size,
-                                    oct_tree=args.mesh_octree_depth,
+                                    oct_tree=max(1, min(14, args.mesh_octree_depth)),
                                     samples=1.0,
                                     point_weight=4.0,
                                     max_concurrency=args.max_concurrency,

--- a/start-dev-env.sh
+++ b/start-dev-env.sh
@@ -55,7 +55,7 @@ if [ "$1" = "--setup" ]; then
 
     # Misc aliases
     echo "alias pdal=/code/SuperBuild/install/bin/pdal" >> $HOME/.bashrc
-    echo "alias opensfm=/code/SuperBuild/src/opensfm/bin/opensfm" >> $HOME/.bashrc
+    echo "alias opensfm=/code/SuperBuild/install/bin/opensfm/bin/opensfm" >> $HOME/.bashrc
     
 
     su -c bash $2

